api/src/pcapi/core/logging.py

```diff
@@ -66,6 +66,24 @@ def get_logged_in_user_id() -> int | None:
         return None
 
 
+def get_logged_impersonator_id() -> int | None:
+    if not _is_within_app_context():
+        return None
+    try:
+        if not current_user:
+            return None
+    except AttributeError:
+        # For some reason, we may get an AttributeError if this
+        # function is called very soon (before some initialization is
+        # done, I guess):
+        #     'Flask' object has no attribute 'login_manager'
+        return None
+    try:
+        return current_user.impersonator.id
+    except AttributeError:  # anonymous use or no impersonator
+        return None
```

Comment:
Je ne suis pas enthousiasmé par le traitement avec une exception, ne serait-ce pas plus clair/propre ainsi ?
```suggestion
    if not current_user.impersonator:
        return None
    return current_user.impersonator.id
```